Azure.Compute.Batch update to 2024-07-01 api

wiboris opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9950 (#48497)

* Use PME for esrp release

* Update to EsrpRelease@9

* Remove extra DomainTenantId parameter

* Add PME service connection to pipelines

---------

Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

wiboris committed:
inital checkin

azure-sdk committed:
Update azure-sdk-build-tools Repository Resource Refs in Yaml files (#48490)

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Compute.Batch](https://apiview.dev/Assemblies/Review/425998d8217d4a3b875a5d0187063ba1?revisionId=e0e95f1e395d4089a266db6f245bf69b)

azure-sdk committed:
Increment package version after release of Azure.AI.Projects (#48496)

blueww committed:
[SRP] Update with latest Swagger (#48465)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Quota (#48509)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Storage (#48511)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DeviceRegistry (#48508)

hallipr committed:
Handle empty yaml variable in interpolated string (#48494)

jsquire committed:
[Governance] Fix outdated references (#48495)

* [Governance] Fix outdated references

The focus of these changes is to bump outdated
references for packages and in code samples that
were being flagged by governance and/or as obsolete
during runs.

Y-Sindo committed:
Prepare release of SignalR extensions 2.0 (#48506)

nickliu-msft committed:
added docs that stream must be seekable for StageBlock (#48515)

KrzysztofCwalina committed:
major CM change (#48487)

scottaddie committed:
Edit pass on Identity README / TSG (#48520)

azure-sdk committed:
Update original file path value in detect change request to APIView (#48519)

Co-authored-by: Praveen Kuttappan <prmarott@microsoft.com>

jsquire committed:
[Event Hubs] Fix trigger data label (#48517)

The focus of these changes is to fix a misspelling of "PartitionId" in the
trigger input data passed to the function executor.  This caused function logs
and metrics reported by AppInsights and the portal to reflect the wrong label.
To ensure that applications that rely on the misspelling are not impacted, a new
member with the correct spelling was added.

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9945 (#48521)

* allow skipping of batching during Create-PRJobMatrix
---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

m-redding committed:
Upgrade to .NET 9 (#48424)

* initial changes

* trying eng sys fix

* update analyzer version

* updates

* tweaks

* tweaks, cleanup, consistency

* attempt

* attempt 2

* testing

* fix

* install .net 9 to unblock generate_matrix

* move the restore test recordings to below the installation of .net

* correct the template yaml

* keyvault fixes

* fix keyvault ci

* keyvault tests

* fix snippet

* attempt 3 at regenerating keyvault snippets...

* try deleting aot proj

* additional pipeline fixes

---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20250303.1 (#48512)

* Update Generator Version 3.0.0-beta.20250303.1

* Update SDK codes co_da_1

---------

Co-authored-by: Wei Hu <huwe@microsoft.com>

ArthurMa1978 committed:
Prepare release for ServiceNetworking (#48386)

scbedd committed:
delete azure.resourcemanager service.props, add a triggering path to core ci so that we will trigger azure.core tests when there are changes to resourcemanager package (#48529)

Allyyyyy committed:
Add platform id to feedback template (#48507)

jsquire committed:
[EngSys] Lock Function extensions dependencies (#48531)

The focus of these changes is to lock the version
of the `Microsoft.Extensions.Azure` version used by
Azure Function extensions packages to 1.8.0, which
is the latest confirmed by the Azure Functions team
to be compatibile with the in-process host.

seanmcc-msft committed:
Updated Storage codeowners (#48533)

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.SignalRService (#48534)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ServiceNetworking (#48538)

amber-ccc committed:
release the first beta version of text authoring (#48335)

* generate first version of new sdk

* add custom files and generate authoring text sdk.

* updated the clientname to add AnalyzeTextAuthoring prefix

* generated the sdk version of shortened names

* remove the gnerated test samples and autogenerate the createProject function

* remove plural from getSubClient function

* remove model subnamespace

* remove`projectName` from Protocol functions of TextAuthoringDeployment

* customize protocol methods for TextAuthoringDeployment and TextAuthoringProject and remove projectname and deploymentName

* Devide the subclient TextAuthoringModel into ExportedModel and TrainedModel

* add convenience method for CreateProject in Text Authoring

* modify the convstrcutor of CreateProjectDetail and remove the projectName parameter

* update all test cases for text authoring

* update Export Api

* Update snippets

* update general readme

* move swapDeployment, assignDeploymentresource and unassignDeploymentresoutce to TextAuthoringProject and modified tests

amber-ccc committed:
Resolving comments for conversation authoring sdk release (#47887)

* modify the clientname to AnalyzeConversationClient and add the AnalyzeConversationClientOptions

* - collapse the client level, only use AnalyzeConversationClient

* suppress `export` with StringIndexType and set it to correct value internally

* -add more subclients.

* modifed all getsubclients methods.

* remove confusionMatrix and confusionRow and use Record<Record<ConfusionCell>> directly

* add one more customed ProjectFilesConversationAuthoringAnalusis and add export and exportAsync in it.

* remove custom getSubClients functions to generate the annotation for use.

* add annotation to all custom methods for successful pack

* changed AnalyzeConversationClient to AnalyzeConversationAuthoringClient

* change prefix from AnalyzeConversation to AnalyzeConversationAuthoring for all common models

* modify files to successfully generate nupkg

* change the subclientname to ConversationAuthoring+ClientletNoun

* change the main client name from AnalyzeConversationAuthoringClient to ConversationAnalysisAuthoringClient

* add prefix 'ConversationAnalysisAuthoring' to main client and clientOption.

* - pass the project name to the subclient constrcutor
- remove client suffix from getSubclient functions

* remove the projectName parameter from all functions of ConversationAuthoringCopyProject

* - remove the projectName parameter from all functions of ConversationAuthoringDeploymentResources

* remove the projectName parameter from all functions of ConversationAuthoringDeployments

* remove the projectName parameter from all functions of ConversationAuthoringExportedModels

* remove the projectName parameter from all functions of ConversationAuthoringModels

* remove ConversationAuthoringPrebuilts constructor with projectname

* remove the projectName parameter from all functions of ConversationAuthoringProjectFiles

* remove ConversationAuthoringProjects constructor with projectname

* remove the projectName parameter from all functions of ConversationAuthoringTraining

* change property `DeploymentExpirationDate` to `DeploymentExpiresOn`

* modify properties type to Azure Regions and Azure Resource Id

* add AssignedDeploymentResource constructor with AzureLocation and ResourceIdentifierId

* change the doc comments of most functions to include the new change of removing ProjectName from functions parameters

* updated ConversationAuthoringCopyProject to ConversationAuthoringCopiedProject, GetCopyProject to GetCopiedProject

* changed the list function to paginated templates

* add one enum type DatasetType for dataset

* Have updated the name to ConversationAnalysisAuthoringModelFactory and move it to .Models namespace.

* Have updated the namespace of Extension to `Microsoft.Extensions.Azure`

* Changed operation name from `copyProjectAuthorization` to `AuthorizeProjectCopy`

* updated parameter name 'body' to 'details'

* adjust the stucture for ConversationAuthoringProject clientlet

* updated the clientlets strcuture and reduce them to three clientlets

* move training mothods to ConversationAuthoringProjects

* move DeploymentResources operations to ConversationAuthoringDeployments

* move copyproject operations to ConversationAuthoringProjects

* moved GetSupportedLanguages and GetProjects to ConversationAnalysisAuthoringClient

* moved GetDeployments and GetAssignedResourceDeployments, etc. to ConversationAnalysisAuthoringClient

* remove suppress on GetAllDeployments, GetAssignedResourceDeployments, ets.

* Move ConversationAuthoringExportedModels into ConversationAuthoringModels

* remove some main client functions from clientlets, to make them generate automatically

* adjust for dotnet pack

* suppress three constructors with no parameters for clientlets

* replace all the parameter `body` with `details`

* replace parameter `body` with `details` for TrainingJobDetails

* remove cache member variables

* modifed the tests based on new version sdk

* updated the commit id for conversations authoring

* updated the readme.file

* revert the changes in text authoring

* update sdk to the latest commit id.

* remove the generated sample tests.

* updated property 'projectName' and `deploymentName` to public

* use the new commit of "removing generated samples"

* updated to the lastest commit id that have been merged to rest api repo

* udpated the export and exportAsync protocol methods

* remove projectname from convenience methods and protocol methods from Project subclient

* remove `projectname` and `deploymentname` from all methods of Deployment subclient

* devide the Models subclient into two and remove the inilialized paramters from methods

* modified the tests based on new subclients and new clientname

* modified all the tests and samples for tests using new clientlets

* updated Export API

* Update snippets

* updated the general readMe

* updated the prefix of common class and updated the related tests

* udpated Export API

* updated Snippets

* update client and clientlets links in general readme

* move swapDeployment, assignDeploymentresource and unassignDeploymentresoutce to TextAuthoringProject and modified tests

* update the links in general readme

* have the type for samples to replace var in all readme files

* replace deploymentAuthoringClient with deploymentClient

* modify the clientlet name in all readme files

KrzysztofCwalina committed:
Added AppConfig and Chat Completion Helper APIs (#48546)

* added support for app configuration

* updated apis

* PR feedback

* PR feedback

* updated api file

azure-sdk committed:
Optimize Service-Level change detection, update ordering of targeted files during impacted package detection (#48548)

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

m-redding committed:
trimming test (#48528)

jsquire committed:
[CODEOWNERS] Fix linter errors (#48550)

The focus of these changes is to remove an owner who is no longer on the team.  Also included are a couple of minor adjustments to Azure SDK team ownership.

scbedd committed:
Remove usage of RemoveTrack1Packages in eng/service.proj (#48514)

* removed an implicit ignore list of packages that are track1. the existing service.projects should classify packages in such a way as to properly include/exclude
* allowed service.projects to flow in regardless of PackagePropsOverrideFile, meaning all projects are properly classified now

JackTn committed:
[SCRIPT] Rewrite .net generate script to write error to devops (#48483)

* wrap errors in the output

* change DotNetSupportChannelLink from donet to dotnet

nickliu-msft committed:
[Storage] [Docs] Added documentation on AggregateException (#48552)

* Added AggregateException docs to each API

* removed test

* removed unused import

christothes committed:
Log the ManagedIdentityId when ManagedIdentityCredential is selected (#48354)

jsquire committed:
[Extensions] Add AzurePipelinesCredential support (#48527)

* [Extensions] Add AzurePipelinesCredential support

The focus of these changes is to add support for specifying that an
AzurePipelinesCredential instance should be used for authorization using
configuration.

amber-ccc committed:
fixed change logs for text authoring and conversation authoring (#48556)

* updated the change log for text authoring

* udpated the change log of conversation authoring

* add `other changes` in change log

v-pivamshi committed:
Added get recording result api. (#47692)

* added get call recording api and Interrupt Hold Audio flag in play options.

* added live test.

* record the live test.

* added live test.

* Fixing the Analyze

* skipping the recording result live test due to current latency to get the recording result is 2 mints

* updaitng the test to test chunks

---------

Co-authored-by: Vinothini Dharmaraj <v-vdharmaraj@microsoft.com>

ShivangiReja committed:
[Azure.AI.project] Implement ConnectionProvider abstraction and add extension methods (#48399)

[Azure.AI.project] Implement ConnectionProvider abstraction and add extension methods

mcgallan committed:
ContainerService Pre-Release (#48542)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ContainerService (#48568)

azure-sdk committed:
Add path loggin to sparse-checkout (#48572)

Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

m-redding committed:
Add comment to Packages.Data.Props for synchronizing SDK upgrades with autorest.csharp (#48571)

paolamvhz committed:
Updating azure rooms SDK for collaborator role (#48200)

* updating for collaborator role

* updating assembly

* updating assembly

* updating generated model

* updating enum

* update api version

* update api version

* update api version

* updating version to 2025-03-13

* updating api version

* updating change log

* updating change log

* updating change log

* removing changes on the constructors

* updating the artifacts

* updating coomunication version for rooms

* updating pragma

* remving extra dot in change log

* adding collaborator tests

* updating samples

m-redding committed:
Add net462 API listings for new target (#48582)

amber-ccc committed:
modify client name of conversation authoring to align with text authoring (#48583)

* modified the clientname to align with Text Authoring

* modifed the tests

* udpated exported api and updated snippets

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9918 (#48551)

* add pullrequest faq

Co-authored-by: Scott Beddall <scbedd@microsoft.com>
Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>
Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

nickliu-msft committed:
Redact SAS credentials from Error.SasCredentialRequiresUriWithoutSas message (#48580)

efrainretana committed:
SDK Microsoft.Search RP 2025-02-01-Preview (#48360)

Commit:
[AzureMonitor] simplify preprocessors (#48553)

* refresh preprocessors

* one more

* additional changes

azure-sdk committed:
Add well known tenant IDs and TME status (#48486)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

nickliu-msft committed:
[Storage] Fixed LazyLoadingReadOnlyStream default buffer size to prevent overprovisioning (#48604)

* Redact SAS credentials from Error.SasCredentialRequiresUriWithoutSas message

* Fixed bug where LazyLoadingReadOnlyStream overprovisions the default buffer memory

* Re-recorded some tests

* Modified changelog

ShivangiReja committed:
Prepare SCM release (#48605)

* Prepare SCM release
* adding logging changes

ckairen committed:
[smoke test] version correction for testing smoke test (#48575)

* trim off beta version for smoke test

* Update eng/scripts/smoketest/Get-Package-Version.ps1

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Update eng/scripts/smoketest/Get-Package-Version.ps1

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Update eng/scripts/smoketest/Get-Package-Version.ps1

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Update eng/scripts/smoketest/Get-Package-Version.ps1

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

---------

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Search (#48590)

azure-sdk committed:
Increment version for cognitivelanguage releases (#48587)

* Increment package version after release of Azure.AI.Language.Conversations.Authoring

* Increment package version after release of Azure.AI.Language.Text.Authoring

Commit:
[AzureMonitor] move LiveMetrics implementation from AspNetCore to LiveMetrics project. (#48498)

* move liveMetrics to new project. update namespaces.

* fix event source references

* cleanup demo project

amnguye committed:
Rerecorded tests after changes to OpenRead default byte value (#48609)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20250306.2 (#48593)

* Update Generator Version 3.0.0-beta.20250306.2

* Update SDK codes he_ma_3

* Update SDK codes da_hd_2

* Update SDK codes pu_sp_5

* Update SDK codes ad_co_0

* Update SDK codes co_da_1

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

---------

Co-authored-by: Wei Hu <huwe@microsoft.com>

ArcturusZhang committed:
remove autorest.md (#48620)

scottaddie committed:
Remove username and password env vars from README (#48626)

christothes committed:
Light up CAE support for ManagedIdentityCredential (#48549)

Commit:
[AzureMonitor] upgrade OTel to v1.11 (#48574)

* upgrade OTel to v1.10 and fix api conflicts

* bugfix: should not create exception telemetry when message is empty

* test bug: fixing bug with requestData.Success

* adding support for net9.0 to csproj and global.json

* upgrade OTel to v1.10 and fix api conflicts

* bugfix: should not create exception telemetry when message is empty

* test bug: fixing bug with requestData.Success

* adding support for net9.0 to csproj and global.json

* cleanup

* bump version 1.11.*

* fix for Http

* changelogs

* changelog

* fix unit tests

* comment

* pr feedback, revert change

KrzysztofCwalina committed:
moved CM connections to app config (#48603)

* moved to app config

* renamed extensions back due to conflick

* updated apis

* added NET9.0 to web project

* added NET9.0 to web.test project

* added NET 462 to web.test project

* removed web tests

* removed web tests

* removed web tests

* removed web tests

azure-sdk committed:
Increment package version after release of System.ClientModel (#48612)

christothes committed:
Prepare for release (#48631)

nickliu-msft committed:
[Storage] Implement IAsyncDisposable in StorageWriteStream (#48628)

* Implemented DisposeAsync in StorageWriteStream

* feedback

ShivangiReja committed:
Update SCM to latest version and change ProjectReference to PackageReference (#48630)

* Update SCM to latest version and change ProjectReference to PackageReference

* Feedback

amnguye committed:
[Storage] [DataMovement] Add Datamovement to Storage CI (#48611)

* [Storage] [DataMovement] Add Datamovement to Storage CI

* Add Blobs to Files tests as well

jalauzon-msft committed:
[Storage][DataMovement] Ensure single call downloads are queued to chunk channel (#48557)
